include/libtorrent/proxy_base.hpp

```diff
@@ -124,6 +124,18 @@ class proxy_base
 		m_sock.async_write_some(buffers, handler);
 	}
 
+#if BOOST_VERSION >= 106600 && !defined TORRENT_BUILD_SIMULATOR
```

Comment:
I imagine I would have to add `async_wait` to uTP connections as well. and I really should add it to libsimulator (but I suppose the simulator doesn't support SSL right now anyway).

I take it this is used by the gnutls wrapper?

Reply:
Yes it is. I observed strange behaviors with null buffers, so since the are now deprecated anyway, I used the recommended `async_wait` which worked fine.